code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -365,37 +363,43 @@ public ProjectMember createProjectMember(Account project, Member member) {
     }
 
     @Override
-    public Account setDefaultProject(Account project, long accountId) {
-        if (project == null) {
-            throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "Failed", "No project specified.", null);
-        }
-        Account account = getAccountById(accountId);
-        if (accountId != project.getId() && project.getKind().equalsIgnoreCase(ProjectConstants.TYPE)
-                && !account.getKind().equalsIgnoreCase(ProjectConstants.TYPE)) {
-            int updateCount = create()
-                    .update(ACCOUNT)
-                    .set(ACCOUNT.PROJECT_ID, project.getId())
-                    .where(ACCOUNT.ID
-                            .eq(accountId))
-                    .execute();
-
-            if (1 != updateCount) {
-                throw new ClientVisibleException(ResponseCodes.NOT_MODIFIED, "Failed", "Failed to update account default project.", null);
+    public void allProjectsHaveNonRancherIdMember(ExternalId externalId) {
```

Comment:
minor: possibly rename to something like "ensureAllProjectsHaveNonRancherIdMembers()".  otherwise, the method signature is a little confusing.